api/napi/src/interpreter/component_compiler.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use std::path::PathBuf;
+
+use super::JsComponentDefinition;
+use super::JsDiagnostic;
+use slint_interpreter::ComponentCompiler;
+
+/// ComponentCompiler is the entry point to the Slint interpreter that can be used
+/// to load .slint files or compile them on-the-fly from a string.
+#[napi(js_name = "ComponentCompiler")]
+pub struct JsComponentCompiler {
+    internal: ComponentCompiler,
+}
+
+#[napi]
+impl JsComponentCompiler {
+    /// Returns a new ComponentCompiler.
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        Self { internal: ComponentCompiler::default() }
+    }
+
+    #[napi(setter)]
+    pub fn set_include_paths(&mut self, include_paths: Vec<String>) {
+        self.internal.set_include_paths(include_paths.iter().map(|p| PathBuf::from(p)).collect());
+    }
+
+    #[napi(getter)]
+    pub fn include_paths(&self) -> Vec<String> {
+        self.internal
+            .include_paths()
+            .iter()
+            .map(|p| p.to_str().unwrap_or_default().to_string())
+            .collect()
+    }
+
+    #[napi(setter)]
+    pub fn set_style(&mut self, style: String) {
+        self.internal.set_style(style);
+    }
+
+    #[napi(getter)]
+    pub fn style(&self) -> Option<String> {
+        self.internal.style().cloned()
+    }
+
+    // todo: set_file_loader
+
+    #[napi(getter)]
+    pub fn diagnostics(&self) -> Vec<JsDiagnostic> {
+        self.internal.diagnostics().iter().map(|d| JsDiagnostic::from(d.clone())).collect()
+    }
+
+    /// Compile a .slint file into a ComponentDefinition
+    ///
+    /// Returns the compiled `ComponentDefinition` if there were no errors.
+    #[napi]
+    pub fn build_from_path(&mut self, path: String) -> Option<JsComponentDefinition> {
+        spin_on::spin_on(self.internal.build_from_path(PathBuf::from(path))).map(|d| d.into())
```

Comment:
I wonder if perhaps we should bite the bullet and enable napi-rs's asyncio or tokio feature and return a Promise to JavaScript. This is supported as a first-class citizen in napi (https://napi.rs/docs/concepts/async-fn ) and it would make our API "correct" I think.

Reply:
The problem is because of `&mut self` for  `napi` this needs to be marked as `unsafe`. I think for now it's ok to do it without a promise for now.